pkg/client/typed/dynamic/client_pool.go

```diff
@@ -46,15 +49,17 @@ type clientPoolImpl struct {
 	config              *restclient.Config
 	clients             map[unversioned.GroupVersion]*Client
 	apiPathResolverFunc APIPathResolverFunc
+	codec               runtime.Codec
 }
 
 // NewClientPool returns a ClientPool from the specified config
-func NewClientPool(config *restclient.Config, apiPathResolverFunc APIPathResolverFunc) ClientPool {
+func NewClientPool(config *restclient.Config, apiPathResolverFunc APIPathResolverFunc, codec runtime.Codec) ClientPool {
```

Comment:
I actually made `dynamic.NewClient` to respect the config. I'll see if I can squeeze the codec into the config.